src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -678,11 +678,12 @@ public function pluck($column, $key = null)
      * @param  array  $columns
      * @param  string  $pageName
      * @param  int|null  $page
+     * @param  array  $options (query, fragment)
      * @return \Illuminate\Contracts\Pagination\LengthAwarePaginator
      *
      * @throws \InvalidArgumentException
      */
-    public function paginate($perPage = null, $columns = ['*'], $pageName = 'page', $page = null)
+    public function paginate($perPage = null, $columns = ['*'], $pageName = 'page', $page = null, array $options = [])
```

Comment:
Do not type-hint `array` for the `$options` parameter
 as we might want to pass an `\ArrayAccess` interface

`public function paginate($perPage = null, $columns = ['*'], $pageName = 'page', $page = null, $options = [])` is just fine